pkg/mcs/scheduling/server/cluster.go

```diff
@@ -170,6 +176,85 @@ func (c *Cluster) SwitchAPIServerLeader(new pdpb.PDClient) bool {
 	return c.apiServerLeader.CompareAndSwap(old, new)
 }
 
+// UpdateScheduler listens on the schedulers updating notifier and manage the scheduler creation and deletion.
+func (c *Cluster) UpdateScheduler() {
+	defer logutil.LogPanic()
+
+	notifier := make(chan struct{}, 1)
+	c.persistConfig.SetSchedulersUpdatingNotifier(notifier)
```

Comment:
If schedulers are initialized and we add/remove a scheduler immediately before setting the notifier, will the notifier lose the update?

Reply:
Fixed. PTAL.